### PR TITLE
Refactor: finish refactor on commands which take optional cell paths.

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

Just scanned through commands which takes optional cell paths, and make more clean up.

And yeah, the refactor doesn't introduce new behavior, just use general `operate` function to simplify code.

The pr also introduce a `ArgumentsCP`(meaned for Arguments only contains cell path) struct, so if some commands only take optional cell paths(like `str length`, `bytes length`), we don't need to create new Arguments struct for these commands.

# Tests + Formatting

Make sure you've done the following, if applicable:

- Add tests that cover your changes (either in the command examples, the crate/tests folder, or in the /tests folder)
  - Try to think about corner cases and various ways how your changes could break. Cover those in the tests

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace --features=extra` to check that all tests pass

# User-Facing Changes

If you're making changes that will affect the user experience of Nushell (ex: adding/removing a command, changing an input/output type, adding a new flag):

- Get another regular contributor to review the PR before merging
- Make sure that there is an entry in the documentation (https://github.com/nushell/nushell.github.io) for the feature, and update it if necessary

### CODE REVIEW

File: crates/nu-command/src/input_handler.rs [hunk 1]
```diff
@@ -7,6 +7,28 @@ pub trait CmdArgument {
     fn take_cell_paths(&mut self) -> Option<Vec<CellPath>>;
 }
 
+/// Arguments with only cell_path.
+///
+/// If commands is going to use `operate` function, and it only required optional cell_paths
+/// Using this to simplify code.
+pub struct ArgumentsCP {
```
Comment (sholderbach) [inline crates/nu-command/src/input_handler.rs:14]: I think the name is a bit obscure: 
Maybe `CellPathArgs` or `CellPathOnlyArgs`?

File: crates/nu-command/src/input_handler.rs [hunk 2]
```diff
@@ -7,6 +7,28 @@ pub trait CmdArgument {
...
+    cell_paths: Option<Vec<CellPath>>,
+}
+
+impl CmdArgument for ArgumentsCP {
+    fn take_cell_paths(&mut self) -> Option<Vec<CellPath>> {
+        self.cell_paths.take()
+    }
+}
+
+impl From<Vec<CellPath>> for ArgumentsCP {
```
Comment (sholderbach) [inline crates/nu-command/src/input_handler.rs:24]: Nice simplification!

File: crates/nu-command/src/conversions/into/datetime.rs [hunk 3]
```diff
@@ -371,7 +349,12 @@ mod tests {
     #[test]
     fn takes_iso8601_date_format() {
         let date_str = Value::test_string("2020-08-04T16:39:18+00:00");
-        let actual = action(&date_str, &None, &None, Span::test_data());
+        let args = Arguments {
+            zone_options: None,
+            format_options: None,
+            cell_paths: None,
+        };
+        let actual = action(&date_str, &args, Span::test_data());
```
Comment (sholderbach) [inline crates/nu-command/src/conversions/into/datetime.rs:357]: A bit unfortunate that the tests have to take the extra structs (so test refactoring with its associated risks is required) but it is much more readable like this, which is nice!

**Review by sholderbach — COMMENTED** [review]
Great work! With the move away from type directed parsing the former `SyntaxShape::CellPath` might require some different treatment. So it is amazing to have this already factored so nicely.

File: crates/nu-command/src/input_handler.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/src/input_handler.rs]: Thanks!  I like the name `CellPathOnlyArgs`

File: crates/nu-command/src/conversions/into/datetime.rs (hunk 3 above)
Comment (WindSoilder) [reply crates/nu-command/src/conversions/into/datetime.rs:357]: Yeah, I've thought about this, because we need to create an extra struct, so the lines of testing code is increased actually, but I think that's ok :)

Comment (sholderbach) [reply crates/nu-command/src/conversions/into/datetime.rs:357]: Looking at a diff in tests is always a bit hard but I think here the benefit of having tests new contributors can quickly read (even though its more lines) outweighs that cost.

**Comment by fdncred** [comment]
Thanks @WindSoilder. I'm also wondering if there are other commands that *should* use `operate` but don't right now. It may be good to have a look through to see where it should be adapted into other commands.

**Comment by WindSoilder** [comment]
> I'm also wondering if there are other commands that should use operate but don't right now. It may be good to have a look through to see where it should be adapted into other commands.

Oh, I've forgot `hash` relative commands.. Just updated it.
And finally checked all and I think there are no more commands can adapt `operate`.  There're some commands doesn't support optionally map data by cell paths, so it don't need to use it

**Comment by WindSoilder** [comment]
I think there are some commands can support map data by cell paths, but currently it didn't, which includes:
* date format
* date humanize
* date to-timezone
* all math relative commands

**Comment by fdncred** [comment]
I just figured there would be a couple handfuls of more commands that should've actually been implemented with an operate type of functionality to be consistent. I was thinking those would be good for another PR.  If there aren't any, that's fine too.

**Review by fdncred — APPROVED** [review]
LGTM! Thanks.

**Comment by dandavison** [comment]
This looks great!

**Review by sholderbach — APPROVED** [review]
